homeassistant/core.py

```diff
@@ -174,6 +174,16 @@ def valid_entity_id(entity_id: str) -> bool:
     return VALID_ENTITY_ID.match(entity_id) is not None
 
 
+def validate_state(state: str) -> str:
```

Comment:
Should we also use this function in the initialiser of `State`?

Reply:
It used to be like that, but @bdraco inlined it so save some cycles.
When it was inlined, that function was however not used outside the constructor of `State`.

@bdraco what's your advice?

Reply:
It only matters on systems with slower clocks that have heavy state churn like ones with many power sensors and at startup when the state machine is being loaded.

We still have ~10% of users on an RPI3 https://analytics.home-assistant.io/#integrations that might feel it. 

tough call but I'd say reuse it in `State` `__init__` init as well since at some point we have to stop worrying about keeping things fast on older hardware as the RPI4 shortage is getting better.